src/utils/MainApi.ts

```diff
@@ -0,0 +1,70 @@
+import { Article } from "../components/App";
+import { BASE_URL } from "./constants";
+import { processServerRes } from "./newsApi";
+
+type updateUserProps = {
+  name: string;
+  avatar: string;
+};
+
+export const updateUser = ({ name, avatar }: updateUserProps) => {
+  const token = localStorage.getItem("jwt");
+  return fetch(`${BASE_URL}/users/me`, {
+    method: "PATCH",
+    headers: {
+      "Content-Type": "application/json",
+      authorization: `Bearer ${token}`,
+    },
+    body: JSON.stringify({
+      name,
+      avatar,
+    }),
+  }).then(processServerRes);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  function request(url, options) {
    return fetch(url, options).then(processServerRes)
  }

```

And now you can safely replace `fetch` with `request` and remove repeated checking. And you don’t need to change anything else.